dags/process_elife_zip_dag.py

```diff
@@ -120,12 +121,14 @@ def convert_tiff_images_in_expanded_bucket_to_jpeg_images(**context) -> None:
                     Key=key,
                     Fileobj=temp_tiff_file
                 )
+                temp_tiff_file.seek(0)
+
+                logger.info('converting %s to jpeg', key)
 
-                # tiff images are typically RGBA
-                # PIL.JpegImagePlugin.RAWMODE contains modes that can be saved as jpeg
-                # In order to convert from tiff we have to remove the alpha channel
                 temp_jpeg = BytesIO()
-                Image.open(temp_tiff_file).convert(mode='RGB').save(temp_jpeg, format='JPEG')
+                with Image(blob=temp_tiff_file) as img:
+                    img.format = 'jpeg'
+                    img.save(file=temp_jpeg)
```

Comment:
`img.destroy()` required?

Reply:
not that I can see in the docs. Also I'm saving to a bytes stream which, being jpeg, should only be in the kilobytes.

Reply:
there's some memory required for reading the TIFF, but not sure how garbage collection will work here without experimenting

Reply:
there's some memory required for reading the TIFF, but not sure how garbage collection will work here without experimentinghttps://sorry-wand.readthedocs.io/en/latest/wand/image.html#wand.image.Image.destroy seems to say calling `destroy` allows garbage collection to work _after_ a context manager

Reply:
the tiff file is a temp file so the context manager should close and delete the file for each tiff after converting.

Reply:
> wand.readthedocs.io/en/latest/wand/image.html#wand.image.Image.destroy

The documentation here says you can call it manually. Looking at the source code, I can see that destroy is called using a context manager.

In wand.image:
```python
class Image(BaseImage):
    def close(self):
        """Closes the image explicitly. If you use the image object in
        :keyword:`with` statement, it was called implicitly so don't have to
        call it.

        .. note::

           It has the same functionality of :attr:`destroy()` method.

        """
        self.destroy()
```

```python
class BaseImage(Resource):
    def destroy(self):
        """Manually remove :class:`~.sequence.SingleImage`'s in
        the :class:`~.sequence.Sequence`, allowing it to
        be properly garbage collected after using a ``with Image()`` context
        manager.

        """
        while self.sequence:
            self.sequence.pop()
        super(Image, self).destroy()
```

```python
class Resource(object):
    def __enter__(self):
        return self

    def __exit__(self, type, value, traceback):
        self.destroy()
```